cmd/webhook/main.go

```diff
@@ -143,19 +144,25 @@ func main() {
 		NodeFlags:            nodeFlags,
 		Namespace:            opt.namespace,
 		VersionConstraints:   constraint,
+
+		// we could change this to get the CertDir from the configured CertName
+		// and KeyName, but doing so does not bring us any benefits but would
+		// technically break compatibility.
+		CertDir:  "/",
+		CertName: opt.admissionTLSCertPath,
+		KeyName:  opt.admissionTLSKeyPath,
 	}.Build()
 	if err != nil {
 		log.Fatalw("Failed to create admission hook", zap.Error(err))
 	}
 
 	log.Infow("Listening", "address", opt.admissionListenAddress)
 
-	if err := srv.ListenAndServeTLS(opt.admissionTLSCertPath, opt.admissionTLSKeyPath); err != nil {
+	serverContext, cancelServerContext := context.WithCancel(context.Background())
```

Comment:
Okay, looks like I do have something to comment on: This shouldn't use `context.Background()` but a context that cancels on SIGINT/SIGTERM so the server shuts down gracefully. Then you can probably drop the `defer cancelServerContext()` as well that I commented on below (don't think we need a `context.WithCancel` then).

You can use  [signals.SetupSignalHandler](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager/signals#SetupSignalHandler) for that (like we do over in the [seed-controller-manager](https://github.com/kubermatic/kubermatic/blob/acf679a2dd5572a6a20168c8b6a493f02afb7cb1/cmd/seed-controller-manager/main.go#L109)).

Reply:
I was thinking about that, but when I run machine-controller locally on my machine, it already reacts perfectly fine to SIGINT - you sure this is needed?

Reply:
Sorry for the late reply. I'm not 100% _sure_ it's needed, but it's the proper way to handle signals and might cover more corner cases than the current code. I would prefer it, if only for consistency with the KKP code base.

Reply:
alright, changed code according to this :)

Sorry for my late reply as well, was still off the first week of the year ^^  
Have a good new year btw :)

Reply:
Happy new year! Hope you got to enjoy the days off :)